sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1231,8 +1231,12 @@ inline std::string GetConnectionKey(std::string const& host, CurlTransportOption
   key.append(!options.SslOptions.EnableCertificateRevocationListCheck ? "1" : "0");
   key.append(options.SslVerifyPeer ? "1" : "0");
   key.append(options.NoSignal ? "1" : "0");
+  // using DefaultConnectionTimeout or 0 result in the same setting
   key.append(
-      options.ConnectionTimeout == CurlTransportOptions::DefaultConnectionTimeout ? "0" : "1");
+      (options.ConnectionTimeout == CurlTransportOptions::DefaultConnectionTimeout
+       || options.ConnectionTimeout == 0)
```

Comment:
Are we sure it is a good idea to make key collisions for this though? I see no benefit, and a slight risk, if things in the future.

Also, I don't think using just 1 here for all the different non-default connection time outs is a good idea. It will cause collisions. I would just simply "int to string" the `ConnectionTimeout` fully, so we get uniqueness.

You don't want the pool to return a connection with a timeout of 50, when you wanted one for 500, right?!

Reply:
> if things change in the future.

We won't be able to change the equivalently of `0 == 300000`.

I will fix the collisions for using 1 and 0

Reply:
> We won't be able to change the equivalently of `0 == 300000`.

Are you sure? How come?

In any case, what's the benefit of the key collisions here? Why not make the code logic simpler and not special case "0"? I imagine different call sites passing different values (0 or 300 000), probably with an implied expectation that the connections are unique.